tests/format/vue/ts-expression/filter.vue

```diff
@@ -0,0 +1,8 @@
+<script setup lang="ts"></script>
+
+<!-- vue filters are only allowed in v-bind and interpolation -->
+<template>
+  <div class="allowed">{{value | thisIsARealSuperLongFilterPipe("arg1", arg2 as unknown) | anotherPipeLongJustForFun | pipeTheThird}}</div>
+  <div class="allowed" v-bind:something='value | thisIsARealSuperLongFilterPipe("arg1", arg2 as unknown) | anotherPipeLongJustForFun | pipeTheThird'></div>
+  <div class="allowed" :class='value | thisIsARealSuperLongFilterPipe("arg1", arg2 as unknown) | anotherPipeLongJustForFun | pipeTheThird'></div>
+</template>
```

Comment:
https://github.com/prettier/prettier/issues/14432 still exists for directives like `v-if`.
But I cannot find a proper parser.
Currently, `__js_expression` is used for Vue directives.
For attribute bindings, I changed parser from `__vue_expression` to `__vue_ts_expression`.
Using `__vue_ts_expression` for directive causes false-positive for Vue filter.

As far as I know, possible approaches are following.
- Approach A: create a new parser `__ts_expression`
- Approach B: leave it until we drop support for Vue filter that is [removed from Vue v3.0](https://v3-migration.vuejs.org/breaking-changes/filters.html)
- Approach C: use `__vue_ts_expression` and somehow detect that `|` is filter or bitwise-or. (I don't know whether it's feasible or not.)

Reply:
Approach A: create a new parser `__ts_expression`

Take this path.

Reply:
Wait, we already have `__vue_ts_expression` it should work.

Reply:
Thanks, I tried to employ `__vue_ts_expression`. https://github.com/prettier/prettier/pull/14506/commits/e057f090e411e79a9cb5c63c586cc92758876205
It almost works correctly.
The problem I mind is that it considers `|` as [Vue filter](https://v2.vuejs.org/v2/guide/filters.html?redirect=true) rather than bitwise-or operator.
It places operator `|` start of lines rather than end of lines.
I would try to create a `__ts_expression` parser if you mind this too.

But I found that `__vue_ts_expression` is better enough than before.

Reply:
We can add node.type check to distinguish, 

```diff
node.operator === "|" &&
+ node.type === "BinaryExpression" && 
        (options.parser === "__vue_expression" ||
          options.parser === "__vue_ts_expression")
```

Reply:
If it's not enough. We can also add parent node check, since vue filter can only use on expression root.

Reply:
Thank you for your replies.
I'm sorry but I don't understand you :dizzy_face:

> We can add node.type check to distinguish

I think it's indistinguishable because the code we pass to parser can be completely identical between vue filter and bitwise-or.
I think this is the reason why current implementation uses `__js_expression` for vue-directive and uses `__vue_expression` for vue-bind.

> If it's not enough. We can also add parent node check, since vue filter can only use on expression root.

I didn't know it. I'll add distinction if you want. But it will be complex a bit because chained filter seems to be a nested binary expression.
However, the original problem of this thread https://github.com/prettier/prettier/pull/14506#discussion_r1136369217 won't be solved.

Feel free to correct me.

Reply:
I think you are right, let's add `__ts_expression` then.

Reply:
Thanks so much, I missed TS union type 👍